From de0410c4694552c137f81b9cbb4926f0d9afeea3 Mon Sep 17 00:00:00 2001
From: chandavath <rcrajagopal@habana.ai>
Date: Sat, 10 Feb 2024 13:37:35 +0200
Subject: [PATCH] [SW-137122] gaudi3: erom rec: fwupd sec weak functions

This patch includes security weak functions to pass through
erom recovery compilation.

Change-Id: Ifc86e05b11713f3c21fab38a56b8fd9654a12a99
Signed-off-by: chandavath <rcrajagopal@habana.ai>
---

diff --git a/lib/preboot/module_fwupd.c b/lib/preboot/module_fwupd.c
index 3fc582a..ad87577 100644
--- a/lib/preboot/module_fwupd.c
+++ b/lib/preboot/module_fwupd.c
@@ -64,6 +64,24 @@
 static struct fwupd_state i2c_state;
 static struct fwupd_resp_prop resp_props[RESP_PROP_NUM];
 
+__weak void preboot_security_dynamic_img_sram_close(void) {}
+__weak void preboot_security_dynamic_img_elbi_reg_open(void) {}
+__weak void preboot_security_dynamic_img_elbi_reg_close(void) {}
+__weak void preboot_security_dynamic_img_sram_open(enum mem_region img_region,
+					enum preboot_mem_access_mode mode)
+{
+}
+
+__weak void preboot_dynamic_img_elbi_reg_cleanup(void *map_addr, u32 size) {}
+__weak int preboot_set_sram_sf_access(bool is_enable)
+{
+	return 1;
+}
+__weak uintptr_t preboot_get_elbi_reg_base(void)
+{
+	return 1;
+}
+
 static u32 fwupd_cmd_readl(void)
 {
 	return regmap_readl(global_conf.upd_cmd_map);
